src/main/groovy/org/gradle/android/workarounds/JdkImageWorkaround.groovy

```diff
@@ -53,16 +53,28 @@ class JdkImageWorkaround implements Workaround {
         applyRuntimeClasspathNormalization(project)
 
         applyToAllAndroidVariants(project) { variant ->
-            variant.javaCompileProvider.configure { JavaCompile task ->
-                def jdkImageInput = getJdkImageInput(task)
-                if (jdkImageInput != null) {
-                    setupExtractedJdkImageInputTransform(project, getJvmHome(task))
-                    replaceCommandLineProvider(task, jdkImageInput)
+            if (Versions.CURRENT_ANDROID_VERSION <= VersionNumber.parse("7.4.0-alpha01")) {
```

Comment:
is there a newer replacement for `variant.javaCompileProvider` in 7.4 + ?

Reply:
sadly nope, https://androidstudygroup.slack.com/archives/C03KKLGC2/p1591411391491800

Reply:
Can we do something like `variant.javaCompilation` like we do in the `RoomWorkaround`?

https://github.com/gradle/android-cache-fix-gradle-plugin/blob/main/src/main/groovy/org/gradle/android/workarounds/RoomSchemaLocationWorkaround.groovy#L116

Reply:
tried yesterday and same issue, at the time the variant is being processed we don't have agp `jdkImageInput` required to apply the workaround.

Reply:
to extend the previous comment, in the previous AGP variant you have access to the Task Provider of Java task, because is lazy we can guarantee that the required ap options have been configured. 
The new variant removes this functionality, in the case of the RoomWorkaround we don't require to have existing ap options because we are just inserting new argument provider in the variant block.

Reply:
@runningcode, following Xavier's comments, there is no replacement for `variant.javaCompileProvider` but we can still use the old API before is removed in AGP 9. 
I created new commit using the old API https://github.com/gradle/android-cache-fix-gradle-plugin/pull/419/commits/fa6672849301f8e69a3fbcb1e5853d341f8a98db and worked

Reply:
That sounds good. Thanks for getting Xav's feedback there.